scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -476,16 +494,28 @@ void BPF_STRUCT_OPS(layered_enqueue, struct task_struct *p, u64 enq_flags)
 		return;
 
 	bpf_for(idx, 0, nr_possible_cpus) {
-		struct cpu_ctx *cand_cctx;
+		struct cpu_ctx *cand_cctx, *sib_cctx = NULL;
 		u32 cpu = (preempt_cursor + idx) % nr_possible_cpus;
+		s32 sib;
 
 		if (!all_cpumask ||
 		    !bpf_cpumask_test_cpu(cpu, (const struct cpumask *)all_cpumask))
 			continue;
+
 		if (!(cand_cctx = lookup_cpu_ctx(cpu)) || cand_cctx->current_preempt)
 			continue;
+		if (layer->exclusive && (sib = sibling_cpu(cpu)) >= 0 &&
+		    (!(sib_cctx = lookup_cpu_ctx(sib)) ||
+		     sib_cctx->current_preempt)) {
+			lstat_inc(LSTAT_EXCL_COLLISION, layer, cctx);
+			continue;
+		}
 
 		scx_bpf_kick_cpu(cpu, SCX_KICK_PREEMPT);
+		if (sib_cctx && !sib_cctx->maybe_idle) {
+			lstat_inc(LSTAT_EXCL_PREEMPT, layer, cctx);
+			scx_bpf_kick_cpu(sib, SCX_KICK_PREEMPT);
+		}
```

Comment:
This whole logic is a bit hard to parse. Some comments could help:

1) `sib_cctx` is only set if `layer->exclusive` so this second condition is really the same as `layer->exclusive && ...`
2) `!sib_cctx->maybe_idle` is still a bit hard for me to follow. I guess the idea here is to kick if there's any chance the sibling could be running something.

Reply:
1. Yeah, so testing for `sib_cctx` is, ignoring the tests which cannot fail but have to be there for verifier, the same as testing `exclusive && if the CPU has a sibling`.
2. If this task is exclusive, we want to make sure the sibling CPU is idle. so if the other CPU is not idle, I kick it w/ PREEMPT so that it expires the current task and enters the dispatch path, where it will see that I'm running an exclusive task and return without dispatching anything forcing it into idle.

Will add comments.